### PR TITLE
Disable preview if post type does not supports it.

### PR DESCRIPTION
**Pull request by jorgefilipecosta**
In the classic editor, the preview button is only available if is_post_type_viewable returns true. We are updating the logic to follow the same behavior.

Fixes: https://github.com/WordPress/gutenberg/issues/5749

## How Has This Been Tested?
Register a post type with publicly_queryable set to false.
E.g:
```
function create_product_type() {
    register_post_type( 'acme_product',
        array(
			'label'  => 'Product',
            'labels' => array(
                'name' => __( 'Products' ),
                'singular_name' => __( 'Product' )
            ),
            'public' => true,
			'show_in_rest' => true,
			'publicly_queryable' => false,
        )
    );
}
add_action( 'init', 'create_product_type' );
```
Verify preview button is not available for posts of this Post Type.

### CODE REVIEW

File: lib/client-assets.php [hunk 1]
```diff
@@ -928,12 +928,15 @@ function gutenberg_editor_scripts_and_styles( $hook ) {
...
 
+	$post_type_object = get_post_type_object( $post_to_edit['type'] );
+
 	$editor_settings = array(
 		'alignWide'           => $align_wide || ! empty( $gutenberg_theme_support[0]['wide-images'] ), // Backcompat. Use `align-wide` outside of `gutenberg` array.
 		'availableTemplates'  => wp_get_theme()->get_page_templates( get_post( $post_to_edit['id'] ) ),
 		'blockTypes'          => $allowed_block_types,
 		'disableCustomColors' => get_theme_support( 'disable-custom-colors' ),
 		'disablePostFormats'  => ! current_theme_supports( 'post-formats' ),
+		'disablePreview'      => ! is_post_type_viewable( $post_type_object ),
```
Comment (youknowriad) [inline lib/client-assets.php]: Can't we retrieve this information using the REST API?

Comment (jorgefilipecosta) [reply lib/client-assets.php]: Unfortunately, it is not available there :(

Comment (aduth) [reply lib/client-assets.php]: Can we create an issue to promote adding this to the REST API?

We're getting a bit too comfortable with cheating where the REST API should be providing necessary functionality.

Comment (danielbachhuber) [reply lib/client-assets.php]: We can include this data on this `/wp-json/wp/v2/types` endpoint.

Comment (jorgefilipecosta) [reply lib/client-assets.php]: Yes including it in the types seems like a good idea.

**Comment by danielbachhuber** [comment]
@jorgefilipecosta Did you create a Trac issue for this? I thought I might pick it up if you aren't already taking care of it.

**Pull request closed without merging**